### DISCUSSION TITLE
Headings containing colon-delimited words don't render properly in the TOC. Docusaurus or Remark issue?

### DISCUSSION THREAD

**Discussion by taylorreece** [facebook/docusaurus#10620] [Q&A] [answered]
In Docusaurus 2.x, headings that contained words delimited by colons (in my example here, `alerts:groups:create`) were rendered in the TOC correctly:

<img width="1199" alt="image" src="https://github.com/user-attachments/assets/ce84801c-67b6-4f7c-8432-30f533d9ab75">

In Docusaurus 3.x (latest build as of 2024-10-26), the final portion of colon-delimited words are removed. So `foo:bar:baz:buz` shows up in the TOC as `foo:bar:baz`.

<img width="1207" alt="image" src="https://github.com/user-attachments/assets/92c137aa-1b39-4114-b58a-2846be524706">

I'm not intimately familiar with the inner workings of either project, but near as I can tell this seems like a remark problem (not a Docusaurus problem). 

Could someone help me confirm that? I need help figuring out which project I should create an issue in (and, hopefully, contribute a solution to).

Thanks!

**Comment by slorber** [comment, marked as answer]
Yes this looks like a remark problem related to toc and directives

:buz is a directive and by default would not show. 

It happens that users often write such things and expect MDX to treat it as text so we added code to convert that back to text, but that conversion happens after the TOC remark extraction. Will need to think about it but in the meantime you can escape `\:buz`

**Comment by taylorreece** [comment]
The `\:buz` escape solution worked great for me. Thanks a bunch!
